### PR TITLE
[M-0011] Add ProxyResultBuilder on handler layer

### PR DESCRIPTION
**Pull request by Aristevao**
**Reviewer**

- @bigLucas.

**Implementation**

- Update all handler files with `ProxyResultBuilder`.

### CODE REVIEW

File: src/api/handlers/PetFindManyHandler.ts [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
 
-    constructor() {
-        this.service = new PetService();
+    public async initializeDependencies(): Promise<void> {
+        const database = new Database();
+        try {
+            await database.connect();
+        } catch (e) {
+            await database.disconnect();
+            throw e;
```
Comment (Aristevao) [inline src/api/handlers/PetFindManyHandler.ts]: @bigLucas 
Adding `ProxyResultBuilder` I needed to instantiate the `Database` to get its `connection`. To solve this, I created the `Database` instance in the same function as the `PetService` instance.
In the PF project, I noticed that it has a `ServerlessDatabase` class and the handler extends that class.
I just made this commitment to show you how it is now.
Do you think I should create a new `ServerlessDatabase` class and follow PF inheritance structure or keep it as now?

Comment (Aristevao) [reply src/api/handlers/PetFindManyHandler.ts]: @bigLucas 
Reading M-0012 carefully, I found that I am doing more than necessary in this branch (M-0011).
Your idea was just to replace the current code inside `execute()` with the new one: `return new ProxyResultBuilder().status (200).body(response).build().`
And handle database stuffs in the next branch (M-0012).
Am I right?

Comment (bigLucas) [reply src/api/handlers/PetFindManyHandler.ts]: @Aristevao 
Yes, you're right!
You can reset the changes and just add the use of the builder.

Comment (bigLucas) [reply src/api/handlers/PetFindManyHandler.ts]: @Aristevao 
You also can add the builder in all files of the handler layer.

File: src/api/handlers/PetDeleteHandler.ts [hunk 2]
```diff
@@ -1,24 +1,17 @@
...
-        const response = {
-            statusCode: 204,
-            headers: {
-                "x-custom-header": "My Header Value",
-                "Access-Control-Allow-Origin": "*",
-                "Access-Control-Allow-Credentials": true,
-            },
-        };
-        return response;
+    public execute(event: any): ProxyResult {
```
Comment (Aristevao) [inline src/api/handlers/PetDeleteHandler.ts:12]: @bigLucas 
Can I already set `event` type as `APIGatewayEvent` instead `any` ?

Comment (bigLucas) [reply src/api/handlers/PetDeleteHandler.ts:12]: @Aristevao yes, you can.

File: src/api/handlers/PetFindManyHandler.ts [hunk 3]
```diff
@@ -1,29 +1,21 @@
...
+import { ProxyResultBuilder } from "../core/ProxyResultBuilder";
 import { PetService } from "../services/PetService";
 
 class PetFindManyHandler {
-    private service: PetService;
+    private petService: PetService;
 
-    constructor() {
-        this.service = new PetService();
+    public initializeDependencies(): void {
```
Comment (bigLucas) [inline src/api/handlers/PetFindManyHandler.ts]: @Aristevao here you can still using the `constructor` instead add this `initializeDependencies()` function.

**Pull request merged**